File: notas.rb
print "Digite sua nota: "
nota = gets.chomp.tr(',', '.').to_f  # Substitui vírgula por ponto e converte para float

conceito = case nota
  when 9.7..10 then "A+"
  when 9.3..9.6 then "A"
  when 9.0..9.2 then "A-"
  when 8.7..8.9 then "B+"
  when 8.3..8.6 then "B"
  when 8.0..8.2 then "B-"
  when 7.7..7.9 then "C+"
  when 7.3..7.6 then "C"
  when 7.0..7.2 then "C-"
  when 6.7..6.9 then "D+"
  when 6.3..6.6 then "D"
  when 6.0..6.2 then "D-"
  when 0..5.9 then "F"
  else "Nota inválida"
end

puts "Nota conceito: #{conceito}"

File: notas.py
print('Escreve sua nota: ')
nota = float(input().replace(',', '.'))
nota = round(nota, 2)
case = {
    0 <= nota <= 5.9: 'F',
    6 <= nota <= 6.2: 'D-',
    6.3 <= nota <= 6.6: 'D',
    6.7 <= nota <= 6.9: 'D+',
    7 <= nota <= 7.2: 'C-',
    7.3 <= nota <= 7.6: 'C',
    7.7 <= nota <= 7.9: 'C+',
    8 <= nota <= 8.2: 'B-',
    8.3 <= nota <= 8.6: 'B',
    8.7 <= nota <= 8.9: 'B+',
    9 <= nota <= 9.2: 'A-',
    9.3 <= nota <= 9.6: 'A',
    9.7 <= nota <= 10: 'A+'
}